packages/cli/src/ui/components/messages/DiffRenderer.tsx

```diff
@@ -235,115 +263,151 @@ const renderDiffContent = (
-            prefixSymbol = '-';
-            // For deletions, update lastLineNumber based on oldLine if it's advancing.
-            // This helps manage gaps correctly if there are multiple consecutive deletions
-            // or if a deletion is followed by a context line far away in the original file.
-            if (line.oldLine !== undefined) {
-              lastLineNumber = line.oldLine;
-            }
-            break;
-          case 'context':
-            gutterNumStr = (line.newLine ?? '').toString();
-            prefixSymbol = ' ';
-            lastLineNumber = line.newLine ?? null;
-            break;
-          default:
-            return acc;
-        }
-
-        const displayContent = line.content.substring(baseIndentation);
+            ) : (
+              // We can use a proper separator when not using max sized box.
+              <Box
+                borderStyle="double"
+                borderLeft={false}
+                borderRight={false}
+                borderBottom={false}
+                width={terminalWidth}
+                borderColor={semanticTheme.text.secondary}
+                marginRight={1}
+              ></Box>
+            )}
+          </Box>,
+        );
+      }
 
-        acc.push(
-          <Box key={lineKey} flexDirection="row">
+      const lineKey = `diff-line-${index}`;
+      let gutterNumStr = '';
+      let prefixSymbol = ' ';
+
+      switch (line.type) {
+        case 'add':
+          gutterNumStr = (line.newLine ?? '').toString();
+          prefixSymbol = '+';
+          lastLineNumber = line.newLine ?? null;
+          break;
+        case 'del':
+          gutterNumStr = (line.oldLine ?? '').toString();
+          prefixSymbol = '-';
+          // For deletions, update lastLineNumber based on oldLine if it's advancing.
+          // This helps manage gaps correctly if there are multiple consecutive deletions
+          // or if a deletion is followed by a context line far away in the original file.
+          if (line.oldLine !== undefined) {
+            lastLineNumber = line.oldLine;
+          }
+          break;
+        case 'context':
+          gutterNumStr = (line.newLine ?? '').toString();
+          prefixSymbol = ' ';
+          lastLineNumber = line.newLine ?? null;
+          break;
+        default:
+          return acc;
+      }
+
+      const displayContent = line.content.substring(baseIndentation);
+
+      const backgroundColor =
+        line.type === 'add'
+          ? semanticTheme.background.diff.added
+          : line.type === 'del'
+            ? semanticTheme.background.diff.removed
+            : undefined;
+      acc.push(
+        <Box key={lineKey} flexDirection="row">
+          {useMaxSizedBox ? (
             <Text
               color={semanticTheme.text.secondary}
+              backgroundColor={backgroundColor}
+            >
+              {gutterNumStr.padStart(gutterWidth)}{' '}
+            </Text>
+          ) : (
+            <Box
```

Comment:
when we aren't using MaxSizedBox we have to do this match the behavior MaxSizedBox provided. Generally this is the idiomatic Ink pattern to ensure that wrapped lines start after this gutter and that this gutter cannot shrink even if flex would otherwise want it to.